packages/@aws-cdk/aws-servicecatalog/lib/tag-options.ts

```diff
@@ -1,14 +1,56 @@
+import * as cdk from '@aws-cdk/core';
+import { hashValues } from './private/util';
+import { InputValidator } from './private/validation';
+import { CfnTagOption } from './servicecatalog.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
 /**
- * Defines a Tag Option, which are similar to tags
- * but have multiple values per key.
+ * Properties for TagOptions.
  */
-export class TagOptions {
+export interface TagOptionsProps {
   /**
-  * List of CfnTagOption
-  */
-  public readonly tagOptionsMap: { [key: string]: string[] };
+   * TagOption keys and associated list of possible values.
+   */
+  readonly tagOptions: { [key: string]: string[] };
+}
+
+/**
+ * Defines a set of TagOptions, which are a list of key-value pairs managed in AWS Service Catalog.
+ * It is not an AWS tag, but serves as a template for creating an AWS tag based on the TagOption.
+ * @resource AWS::ServiceCatalog::TagOption
+ */
+export class TagOptions extends cdk.Resource {
+  /**
+   * Map of underlying TagOption resources.
+   *
+   * @internal
+   */
+  public readonly _tagOptionsMap: { [key: string]: CfnTagOption };
+
+  constructor(scope: Construct, id: string, props: TagOptionsProps) {
+    super(scope, id);
+    this._tagOptionsMap = this.createUnderlyingTagOptions(props.tagOptions);
```

Comment:
Because we iterate over the underlying values for creation and associating, if you do an empty one it will not fail and it won't create anything.  I can add a test for this case as well, or since this it's mostly useless we could validate and fail.

Reply:
It's better for this to fail since it wouldn't align with other constructs to instantiate one with a certain (empty) configuration that wouldn't yield anything in template.  I've therefore added validation to make sure that the struct is not empty, and then that for each key the tag values are not empty, along with two tests for this.  There is not a max length for these lists since I believe that these can be altered by the customer.